src/libcollections/slice.rs

```diff
@@ -691,15 +691,35 @@ impl<T> [T] {
     ///
     /// # Examples
     ///
-    /// Print the slice split by numbers divisible by 3 (i.e. `[10, 40]`,
-    /// `[20]`, `[50]`):
+    /// ```
+    /// let slice = [10, 40, 33, 20];
+    /// let mut iter = slice.split(|num| num % 3 == 0);
+    /// assert_eq!(iter.next().unwrap(), &[10, 40]);
+    /// assert_eq!(iter.next().unwrap(), &[20]);
+    /// assert!(iter.next().is_none());
+    /// ```
+    ///
+    /// If the first or last element in the slice is matched, an empty slice
```

Comment:
I'm not too satisfied with the sentence I wrote. I basically wanted to indicate that if the matched element is the first item in the slice, then an empty slice will be returned from the iterator first. Similarly, if the last element is matched, the last item the iterator returns is an empty slice. I was trying to combine that in one sentence.

In case you're not familiar with the word "respectively": http://english.stackexchange.com/questions/138695/when-to-use-respectively